### PR TITLE
SW-651 Scripts and styles for the no-ui-slider must only be loaded on search and category pages

### PR DESCRIPTION
**Pull request by nenadj-soprex**
Struct PageInformation add to pagelet extension, condition for no-ui-slider

### CODE REVIEW

File: src/Subscriber/FrontendSubscriber.php [hunk 1]
```diff
@@ -78,5 +79,13 @@ public function onHeaderLoaded(HeaderPageletLoadedEvent $event): void
 
         // Save the snippet for usage in template
         $event->getPagelet()->addExtension('flSnippet', $snippet);
+
+        $request = $event->getRequest();
+        $isSearchPage = $request->query->get('search') !== null && str_contains($request->getRequestUri(), '/search');
```
Comment (TobiasGraml11) [inline src/Subscriber/FrontendSubscriber.php]:
```suggestion
        $isSearchPage = $request->query->has('search') && str_contains($request->getRequestUri(), '/search');
```

File: src/Subscriber/FrontendSubscriber.php [hunk 2]
```diff
@@ -78,5 +79,13 @@ public function onHeaderLoaded(HeaderPageletLoadedEvent $event): void
...
+        $isNavigationPage = boolval($request->attributes->get('navigationId'));
```
Comment (TobiasGraml11) [inline src/Subscriber/FrontendSubscriber.php]:
```suggestion
        $isNavigationPage = $request->attributes->has('navigationId');
```

File: tests/Subscriber/FrontendSubscriberTest.php [hunk 3]
```diff
@@ -54,6 +83,15 @@ public function testHeaderPageletLoadedEvent(): void
...
             ->getMock();
 
+        $request = new Request(
+            $requestParams[0],
+            $requestParams[1],
+            $requestParams[2],
+            $requestParams[3],
+            $requestParams[4],
+            $requestParams[5]
+        );
```
Comment (TobiasGraml11) [inline tests/Subscriber/FrontendSubscriberTest.php]:
```suggestion
        $request = new Request(... $requestParams);
```

File: tests/Subscriber/FrontendSubscriberTest.php [hunk 4]
```diff
@@ -34,10 +28,45 @@ class FrontendSubscriberTest extends TestCase
...
+                    [],
+                    [],
+                    [],
+                    ['REQUEST_URI' => 'https://example.com/search']
+                ],
+                'expectedPageInformation' => [
+                    'isSearchPage' => true,
+                    'isNavigationPage' => false
+                ]
+            ],
```
Comment (TobiasGraml11) [inline tests/Subscriber/FrontendSubscriberTest.php:89]: Please add test examples for all cases. (e.g. request url includes "/search", but search parameter is not given).

Would be great, if you could split the test cases into a new test function. In this function, you could just skip the check for `PageInformation`.

**Review by TobiasGraml11 — CHANGES_REQUESTED** [review]

File: tests/Subscriber/FrontendSubscriberTest.php (hunk 3 above)
Comment (nenadj-soprex) [reply tests/Subscriber/FrontendSubscriberTest.php]: @TobiasGraml11 thank you for this syntax :)

**Review by TobiasGraml11 — APPROVED** [review]

**Pull request merged**